sdk/ml/azure-ai-ml/azure/ai/ml/operations/_evaluator_operations.py

```diff
@@ -0,0 +1,772 @@
+module_logger = ops_logger.module_logger
+
+
+class EvaluatorOperations(_ScopeDependentOperations):
+    """EvaluatorOperations.
+
+    You should not instantiate this class directly. Instead, you should create an MLClient instance that instantiates it
+    for you and attaches it as an attribute.
+
+    :param operation_scope: Scope variables for the operations classes of an MLClient object.
+    :type operation_scope: ~azure.ai.ml._scope_dependent_operations.OperationScope
+    :param operation_config: Common configuration for operations classes of an MLClient object.
+    :type operation_config: ~azure.ai.ml._scope_dependent_operations.OperationConfig
+    :param service_client: Service client to allow end users to operate on Azure Machine Learning Workspace
+        resources (ServiceClient082023Preview or ServiceClient102021Dataplane).
+    :type service_client: typing.Union[
+        azure.ai.ml._restclient.v2023_04_01_preview._azure_machine_learning_workspaces.AzureMachineLearningWorkspaces,
+        azure.ai.ml._restclient.v2021_10_01_dataplanepreview._azure_machine_learning_workspaces.
+        AzureMachineLearningWorkspaces]
+    :param datastore_operations: Represents a client for performing operations on Datastores.
+    :type datastore_operations: ~azure.ai.ml.operations._datastore_operations.DatastoreOperations
+    :param all_operations: All operations classes of an MLClient object.
+    :type all_operations: ~azure.ai.ml._scope_dependent_operations.OperationsContainer
+    """
+
+    # pylint: disable=unused-argument
+    def __init__(
+            self,
+            operation_scope: OperationScope,
+            operation_config: OperationConfig,
+            service_client: Union[ServiceClient082023Preview, ServiceClient102021Dataplane],
+            datastore_operations: DatastoreOperations,
+            all_operations: Optional[OperationsContainer] = None,
+            **kwargs: Dict,
+    ):
+        super(EvaluatorOperations, self).__init__(operation_scope, operation_config)
+        ops_logger.update_info(kwargs)
+        self._model_versions_operation = service_client.model_versions
+        self._model_container_operation = service_client.model_containers
+        self._service_client = service_client
+        self._datastore_operation = datastore_operations
+        self._all_operations = all_operations
+        self._control_plane_client: Any = kwargs.get("control_plane_client", None)
+        self._workspace_rg = kwargs.pop("workspace_rg", None)
+        self._workspace_sub = kwargs.pop("workspace_sub", None)
+        self._registry_reference = kwargs.pop("registry_reference", None)
+
+        # Maps a label to a function which given an asset name,
+        # returns the asset associated with the label
+        self._managed_label_resolver = {"latest": self._get_latest_version}
+
+    @monitor_with_activity(ops_logger, "Evaluator.CreateOrUpdate", ActivityType.PUBLICAPI)
+    def create_or_update(  # type: ignore
```

Comment:
Do you plan to introduce the noun for "Evaluator" at control plane side for build? If that's not the plan for short term, is it possible to reuse existing model APIs for evalutor registration?

Reply:
In this PR we are mostly re-using model API with some addons. Generally, our evaluator represent artifacts like the models, however we do not want to mix these two concepts.